tests/api/test_collect_runs_logs.py

```diff
@@ -166,6 +168,163 @@ async def test_collect_logs_on_startup(
             == {}
         )
 
+    @pytest.mark.asyncio
+    async def test_collect_logs_for_old_runs_on_startup(
+        self,
+        db: sqlalchemy.orm.session.Session,
+        client: fastapi.testclient.TestClient,
+    ):
+        log_collector = mlrun.api.utils.clients.log_collector.LogCollectorClient()
+
+        project_name = "some-project"
+        new_uid = "new_uid"
+        old_uid = "old_uid"
+
+        # create first run
+        _create_new_run(
+            db,
+            project_name,
+            uid=old_uid,
+            name=old_uid,
+            kind="job",
+            state=mlrun.runtimes.constants.RunStates.completed,
+        )
+
+        # sleep for 5 seconds to make sure the runs are not created at the same time
+        time.sleep(5)
+
+        # create second run
+        _create_new_run(
+            db,
+            project_name,
+            uid=new_uid,
+            name=new_uid,
+            kind="job",
+            state=mlrun.runtimes.constants.RunStates.completed,
+        )
+
+        # verify that we have 2 runs
+        runs = mlrun.api.utils.singletons.db.get_db().list_distinct_runs_uids(
+            db,
+            requested_logs_modes=[False],
+            only_uids=False,
+        )
+        assert len(runs) == 2
+
+        # change mlrun config so that the old run will be considered as old
+        previous_grace_period = (
+            mlrun.config.config.runtime_resources_deletion_grace_period
+        )
```

Comment:
this is not really needed. tests.common_fixtures.config_test_base should take care of it